### ISSUE TITLE
GraphQLExecutionContext does not inherits methods from ExecutionContext

### ISSUE THREAD

**Issue by Davide-Gheri** [nestjs/graphql#504]
## I'm submitting a...
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
The interface `GraphQLExecutionContext` implements `ExecutionContext` that exposes some methods to work with the context, ex. `getHandler()` and `getClass()`, but the Object created with `GqlExecutionContext.create()` does not really include them.
I think it is because the `create()` method just do an Object.assign that does not copy the `host.__proto__`, thus forgetting about all the `ExecutionContextHost.prototype` methods.
Now, to access the above mentioned methods I need to call the ExecutionContextHost directly.

**Given that, the current `GraphQLExecutionContext` interface implementation is wrong.**

## Expected behavior
It would be nice to have all the `ExecutionContext` methods in `GraphQLExecutionContext`, so I don't have to keep track of the original ExecutionContext to call the other methods

## Minimal reproduction of the problem with instructions
Just try to call one of the `ExecutionContext` methods from the returned object of `GqlExecutionContext.create(context)`
```
...
canActivate(context: ExecutionContext) {
  const gqlContext = GqlExecutionContext.create(context);

  const handler = gqlContext.getHandler() //TypeError: gqlContext.getHandler is not a function
  const classRef = gqlContext.getClass() //TypeError: gqlContext.getClass is not a function
  const httpArgHost = gqlContext.switchToHttp() // TypeError: gqlContext.switchToHttp is not a function
}
```

## Environment

<pre><code>
Nest version: 6.7.2
Nest graphql version: 6.5.3

 
For Tooling issues:
- Node version: v12.13.1
- Platform:  Mac

</code></pre>

**Comment by jmcdo29** [comment]
Couldn't you still use the `context` variable that was used to create the `gqlContext` and still have access to the methods you want?

**Comment by Davide-Gheri** [comment]
Yes of course, but as I stated before, the `GraphQLExecutionContext` Interface is wrong, it should provide all the methods of the default `ExecutionContext` but they are not there

**Comment by jmcdo29** [comment]
Sure enough, missed that.

**Comment by Davide-Gheri** [comment]
I can open a PR to fix it if you want, what do you think is the best solution? Removing the ExecutionContext methods from GraphQLExecutionContext interface or really implement them?

**Comment by kamilmysliwiec** [comment]
Creating a class that inherits from `ExecutionContext` would be the best I guess. PRs are welcome

**Comment by kamilmysliwiec** [comment]
It should be fixed in 6.5.4 :) Let me know if you face any issues
